docs/core/diagnostics/event-counters.md

```diff
@@ -0,0 +1,353 @@
+
+```csharp
+public class SimpleEventListener : EventListener
+{
+    private readonly EventLevel _level = EventLevel.Verbose;
+    private int _intervalSec;
+
+    public int EventCount { get; private set; }
+
+    public SimpleEventListener(int intervalSec)
+    {
+        _intervalSec = intervalSec;
+    }
+
+    protected override void OnEventSourceCreated(EventSource source)
+    {
+        if (source.Name.Equals("System.Runtime"))
+        {
+            var refreshInterval = new Dictionary<string, string>()
+            {
+                ["EventCounterIntervalSec"] = "1"
+            };
+
+            EnableEvents(source, _level, (EventKeywords)(-1), refreshInterval);
+        }
+    }
+
+    private (string counterName, string counterValue) GetRelevantMetric(
+        IDictionary<string, object> eventPayload)
+    {
+        string counterName = "";
+        string counterValue = "";
+
+        foreach ((string key, object value) in eventPayload)
+        {
+            if (key.Equals("DisplayName"))
+            {
+                counterName = value.ToString();
+            }
+            else if (key.Equals("Mean") || key.Equals("Increment"))
+            {
+                counterValue = value.ToString();
+            }
+        }
+
+        return (counterName, counterValue);
+    }
+
+    protected override void OnEventWritten(EventWrittenEventArgs eventData)
+    {
+        if (eventData.EventName.Equals("EventCounters"))
+        {
+            for (int i = 0; i < eventData.Payload.Count; i++)
+            {
+                if (eventData.Payload[i] is IDictionary<string, object> eventPayload)
+                {
+                    (string counterName, string counterValue) = GetRelevantMetric(eventPayload);
+                    Console.WriteLine($"{counterName} : {counterValue}");
+                }
+            }
+        }
+    }
+}
+```
+
+As shown above, you _must_ make sure the `"EventCounterIntervalSec"` argument is set in the `filterPayload` argument when calling <xref:System.Diagnostics.Tracing.EventListener.EnableEvents%2A>. Otherwise the counters will not be able to flush out values since it doesn't know at which interval it should be getting flushed out.
+
+### Consuming out-of-proc
+
+Consuming EventCounters out-of-proc is also possible. You can use ETW to capture counter data as events, and view them on your ETW trace viewer (PerfView and WPA). You may also use [dotnet-counters](dotnet-counters.md) to consume them in a cross-platform manner via an EventPipe. Likewise, you could also use <xref:System.Diagnostics.TraceSource.TraceEvent%2A?displayProperty=nameWithType>, or <xref:System.Diagnostics.TraceListener.TraceEvent%2A?displayProperty=nameWithType> to consume EventCounters.
```

Comment:
Its true that PerfView or WPA would be able to collect the raw events, but the viewing experience isn't necessarily good (I assume WPA doesn't have any special knowledge of the events or the ability to graph them). I'm not sure its worth the mention.

The links to TraceSource and TraceListener here are not related to the [TraceEvent nuget library](https://www.nuget.org/packages/Microsoft.Diagnostics.Tracing.TraceEvent) that I think you meant to refer to? 

We might want to distinguish three different layer of technology being discussed here so users can understand how the stack fits together. 
- Transporting events in a raw stream - ETW or EventPipe. ETW APIs come with windows OS, EventPipe is accessible via an [API .Net provides](https://github.com/dotnet/diagnostics/blob/master/documentation/design-docs/diagnostics-client-library.md#1-attaching-to-a-process-and-dumping-out-all-the-runtime-gc-events-in-real-time-to-the-console) or there is [a protocol](https://github.com/dotnet/diagnostics/blob/master/documentation/design-docs/ipc-protocol.md) for people who want to get into the weeds.
- Decoding the binary event stream into events - The TraceEvent library handles both ETW and EventPipe stream formats
- Command-line and GUI tools - These are tools like PerfView (ETW or EventPipe), dotnet-counters (EventPipe only), dotnet-monitor (EventPipe only), and soon Visual Studio.